openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -239,6 +264,63 @@ def export(self, request, **_kwargs) -> HttpResponse:
 
         return HttpResponse(tags, content_type=content_type)
 
+    @action(detail=False, url_path="import", methods=["post"])
+    def create_import(self, request: Request, **_kwargs) -> HttpResponse:
+        """
+        Create a new taxonomy and imports the tags from the uploaded file.
+        """
+        perm = "oel_tagging.import_taxonomy"
+        if not request.user.has_perm(perm):
+            raise PermissionDenied("You do not have permission to import taxonomies")
+
+        body = TaxonomyImportNewBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        taxonomy_name = body.validated_data["taxonomy_name"]
+        taxonomy_description = body.validated_data["taxonomy_description"]
+        file = body.validated_data["file"].file
+        parser_format = body.validated_data["parser_format"]
+
+        taxonomy = create_taxonomy(taxonomy_name, taxonomy_description)
+        try:
+            import_success = import_tags(taxonomy, file, parser_format)
+
+            if import_success:
+                return HttpResponse(status=200)
```

Comment:
We should return the created Taxonomy here (serialized with `TaxonomySerializer`) on success.
Then you can also test this response in your view tests, instead of fetching the created instance by taxonomy name.

Reply:
Good catch @pomegranited!
Done: 62e2593e176d2aa4468c49047c0853322e0ffe48

PS: The `Tag` list is not serialized with the `Taxonomy` object. We need a different endpoint for that. I updated the tests to use this endpoint  (as we discussed in another PR) here: a827178ccc74d834ed4dbe085da83d04bcd6b598